storage/repository/generic.go

```diff
@@ -122,7 +157,111 @@ func GenericCopy(source fyne.URI, destination fyne.URI) error {
 // NOTE: this function should not be called except by an implementation of
 // the Repository interface - using this for unknown URIs may break.
 //
-// Since 2.0.0
+// Since: 2.0.0
 func GenericMove(source fyne.URI, destination fyne.URI) error {
-	return fmt.Errorf("TODO")
+	// This looks a lot like GenericCopy(), but I duplicated the code
+	// to avoid having to look up the repositories more than once.
+
+	// Look up repositories for the source and destination.
+	srcrepo, err := ForURI(source)
+	if err != nil {
+		return err
+	}
+
+	dstrepo, err := ForURI(destination)
+	if err != nil {
+		return err
+	}
+
+	// The source and destination must both be writable, since the source
+	// is being deleted, which requires WriteableRepository.
+	destwrepo, ok := dstrepo.(WriteableRepository)
+	if !ok {
+		return OperationNotSupportedError
+	}
+
+	srcwrepo, ok := srcrepo.(WriteableRepository)
+	if !ok {
+		return OperationNotSupportedError
+	}
+
+	// Create the reader and writer to perform the copy operation.
+	srcReader, err := srcrepo.Reader(source)
+	if err != nil {
+		return err
+	}
+
+	dstWriter, err := destwrepo.Writer(destination)
+	if err != nil {
+		return err
+	}
+
+	// Perform the copy.
+	_, err = io.Copy(dstWriter, srcReader)
+	if err != nil {
+		return err
+	}
+
+	// Finally, delete the source only if the move finished without error.
+	return srcwrepo.Delete(source)
+}
+
+// ParseURI implements the back-end logic for storage.ParseURI, which you
+// should use instead. This is only here because other functions in repository
+// need to call it, and it prevents a circular import.
+//
+// Since: 2.0.0
+func ParseURI(s string) (fyne.URI, error) {
```

Comment:
Should this check for `CanonicalRepository` as it is used for this reason by line 97 above?

Reply:
Yes, good catch!

7d75419f93eea1413d8daed37ab9f02848e0343e